app/questionnaire/router.py

```diff
@@ -305,10 +278,16 @@ def get_next_block_url(location, routing_path):
             list_item_id=routing_path.list_item_id,
         )
 
-    def get_last_location_in_survey(self):
-        last_section_id = self._schema.get_section_ids()[-1]
-        last_block_id = self._schema.get_last_block_id_for_section(last_section_id)
-        return Location(section_id=last_section_id, block_id=last_block_id)
+    def get_last_location_in_questionnaire(self) -> Location:
+        section_id, list_item_id = self._get_last_complete_section_key()
+        last_complete_block = self._progress_store.get_completed_block_ids(
+            section_id=section_id, list_item_id=list_item_id
+        )[-1]
```

Comment:
I don't think this will always work as a block can be completed, but not on the routing path.

Reply:
🙈 Good spot. Will refactor.

Reply:
This now returns the last block on the path by generating a routing path.